### PR TITLE
Bicaws 3134 update exception and trigger handlers filter behaviour

### PR DESCRIPTION
**Pull request by C-gyorfi**
### Context
During UR, we found that the combination of filters can result in odd behaviour, specifically combining resolution status, reason code and user permissions. 

### Changes
- Refactor list court cases function to ensure users can filter and see cases that their concern and have permission to see
- Extracted a `filterByReasonAndResolutionStatus` function to generate conditions based on user permission and applied reason/resolution status filter and cross-test these:
  - These filters are tightly coupled as applying one affects the others. For instance, a trigger handler filtering for resolved cases should see cases with resolved triggers only
- Added a test matrix for testing the combination of user permission, case state and reason filters
- Moved test from `listCourtCases` integration tests to test matrix where it makes sense

### Guidance for review
- I'd recommend looking through the test matrix to see if the filtering behaviour makes sense

### CODE REVIEW

File: test/services/listCourtCases/listCourtCases.integration.test.ts
Comment (C-gyorfi) [inline test/services/listCourtCases/listCourtCases.integration.test.ts:1]: I think we can also remove 
`describe("Filter cases having reason", () => {`
from this file

Comment (C-gyorfi) [reply test/services/listCourtCases/listCourtCases.integration.test.ts:1]: `Filter cases by case state` also covered/can be covered in the new test matrix

File: test/services/listCourtCases/listCourtCases.integration.test.ts [hunk 1]
```diff
@@ -933,275 +922,6 @@ describe("listCourtCases", () => {
...
-      expect(isError(result)).toBeFalsy()
-      const { result: cases } = result as ListCourtCaseResult
-
-      expect(cases).toHaveLength(3)
-      expect(cases[0].errorId).toBe(2)
-      expect(cases[1].errorId).toBe(3)
-      expect(cases[2].errorId).toBe(4)
-    })
-
-    it("Should filter cases with all reasons", async () => {
```
Comment (C-gyorfi) [inline test/services/listCourtCases/listCourtCases.integration.test.ts:1031]: Can we also test this in the test matrix (multiple reason options selected)?

File: test/services/listCourtCases/filterByResolutionStatus.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+        filters: {
+          caseState: "Resolved",
+          reasonCode: "TRP"
+        },
+        user: exceptionHandler,
+        expectedCases: ["Exceptions Resolved by exceptionHandler/Trigger Resolved by triggerHandler"]
+      },
+      {
+        description: "Should only see unresolved triggers when case state is not set as a trigger handler",
+        filters: {
```
Comment (C-gyorfi) [inline test/services/listCourtCases/filterByResolutionStatus.integration.test.ts:645]: In this case we could just set filters: {}

File: test/services/listCourtCases/filterByResolutionStatus.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+        expectedCases: [
+          "Exceptions Resolved by exceptionHandler/Trigger Unresolved",
+          "Exceptions Unresolved/Trigger Unresolved",
+          "No exceptions/Bails Trigger Unresolved",
+          "Exceptions Unresolved/Bails Trigger Unresolved"
+        ]
+      },
+      {
+        description: "Should only see unresolved exceptions when case state is not set as an exceptions handler",
+        filters: {
```
Comment (C-gyorfi) [inline test/services/listCourtCases/filterByResolutionStatus.integration.test.ts:659]: Same here

**Review by emadkaramad — APPROVED** [review]

**Pull request merged**